src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,56 @@
+package core.basesyntax;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.handler.BalanceOperationHandler;
+import core.basesyntax.handler.PurchaseOperationHandler;
+import core.basesyntax.handler.ReturnOperationHandler;
+import core.basesyntax.handler.ShopOperationHandler;
+import core.basesyntax.handler.SupplyOperationHandler;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.CsvReaderService;
+import core.basesyntax.service.CsvWriterService;
+import core.basesyntax.service.DataProcessorService;
+import core.basesyntax.service.DataReaderService;
+import core.basesyntax.service.DataWriterService;
+import core.basesyntax.service.FruitShopDataProcessorService;
+import core.basesyntax.strategy.ShopOperationStrategy;
+import core.basesyntax.strategy.ShopOperationStrategyImpl;
+import core.basesyntax.utility.FruitQuantityChecker;
+import core.basesyntax.utility.Operation;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    public static final String INPUT_CSV = "src/main/resources/input.csv";
+    public static final String REPORT_CSV = "src/main/resources/report.csv";
+
+    public static void main(String[] args) {
+        Map<Operation, ShopOperationHandler> shopOperationHandlerMap = new HashMap<>();
+        shopOperationHandlerMap.put(Operation.BALANCE,
+                new BalanceOperationHandler());
+        shopOperationHandlerMap.put(Operation.RETURN,
+                new ReturnOperationHandler());
+        shopOperationHandlerMap.put(Operation.SUPPLY,
+                new SupplyOperationHandler());
+        shopOperationHandlerMap.put(Operation.PURCHASE,
+                new PurchaseOperationHandler());
+
+        Storage.FRUIT_STORAGE.put("banana", new Fruit());
+        Storage.FRUIT_STORAGE.put("apple", new Fruit());
```

Comment:
I cannot accept solution that works only for bananas and apples. I don't know if I need to add oranges or lemons tomorrow.
You solution should accept any proper structured file with any fruit names.
Please rework not to have NPE when any other fruit name is contained in file.

I again recommend to use Map<String, Integer> as it's simplest solution.
But if you wish to stick to your Fruit model with it's methods - then enchance your handlers to create proper record to the map in case such fruit name is not present yet

Reply:
Done